### PR TITLE
Adds targets API route per country

### PR DESCRIPTION
**Pull request by simaob**
This PR creates a JSON endpoint to return the Targets for a given country.

Usage:
http://localhost:3000/cclow/api/targets/:iso

It copies the same formatting as the old one: Example here: https://www.climatewatchdata.org/api/v1/lse_laws_and_policies/BRA

I'm a bit tired and confused, so might have written the most ugly code ever. I had to pass the request protocol and host to get the right path, but maybe I could have added some helper to the model to access it directly. Thoughts? Thanks!

### CODE REVIEW

File: app/models/target.rb [hunk 1]
```diff
@@ -71,4 +71,24 @@ def to_s
...
+      doc_type: source,
+      type: target_type,
+      scope: scopes.map(&:name).join(', '),
+      sector: sector&.name,
+      description: description,
+      sources: legislations.map do |l|
+        {
+          id: l.id,
+          title: l.title,
+          link: "#{path}/cclow/geographies/#{geography.id}/#{l.route}/#{id}"
```
Comment (tsubik) [inline app/models/target.rb]: Rails.application.routes.url_helpers could help with taht

**Comment by tsubik** [comment]
I think I would move that code from model to controller. As this json formar is more specific to only that controller

File: app/models/target.rb (hunk 1 above)
Comment (simaob) [reply app/models/target.rb]: gonna move this to the controller.

**Comment by simaob** [comment]
updated and added specs as a bonus!

**Pull request merged**